R/labs.R

```diff
@@ -0,0 +1,103 @@
+#' @title Easily add ggplot labels using label attribtute of `data.frame` column
+#' @description Applies same logic as \code{\link[ggplot2]{labs}} but uses as default
+#' the column label attribute if present as the variable label in the plot.
+#' @param ... A list of new name-value pairs. The name should either be an aesthetic, 
+#' or one of "title", "subtitle", or "caption"
+#' @return NULL
+#' @examples 
+#' 
+#' require(ggplot2)
+#' iris_labs <- iris
+#' lbl <- c('Sepal Length', 'Sepal Width', 'Petal Length', 'Petal Width', 'Flower Species')
+#' labelled::var_label(iris_labs) <- split(lbl,names(iris_labs))
+#' 
+#' p <- ggplot(iris_labs,aes(x=Sepal.Length,y=Sepal.Width))+
+#'   geom_line(aes(colour=Species))
+#'   
+#' p
+#' p + easy_labs()
+#' p + easy_labs(subtitle='mysubtitle', x='x axis label')
+#'  
+#' @rdname easy_labs
+#' @export 
+easy_labs <- function(...){
+
+  man_labs <- labs(list(...))
+  
+  structure(man_labs, class = "easy_labs")
+  
+}
+
+easy_update_labels <- function(p,man_labs){
+
+  root_dat_labs <- sapply(p$data,attr,which='label')
+  root_dat_labs <- root_dat_labs[!sapply(root_dat_labs,is.null)]
```

Comment:
Consider `non_null_root_dat_labs <- Filter(Negate(is.null), root_dat_labs)` as an alternative, which may be easier to read.